astropy/io/fits/hdu/hdulist.py

```diff
@@ -745,15 +751,29 @@ def append(self, hdu):
                 # so create an Extension HDU from the input Primary HDU.
                 # TODO: This isn't necessarily sufficient to copy the HDU;
                 # _header_offset and friends need to be copied too.
-                hdu = ImageHDU(hdu.data, hdu.header)
+                hdu = ImageHDU(
+                    hdu.data,
+                    hdu.header,
+                    do_not_scale_image_data=hdu._do_not_scale_image_data,
+                )
+                if hdu._do_not_scale_image_data:
+                    hdu.header["BZERO"] = bzero
+                    hdu.header["BSCALE"] = bscale
```

Comment:
Is it possible to get into this block without L740-L743 above and hit undefined variables error?